pytorch_lightning/trainer/data_loading_mixin.py

```diff
@@ -15,8 +18,11 @@ class TrainerDataLoadingMixin(object):
     def layout_bookeeping(self):
 
         # determine number of training batches
-        self.nb_training_batches = len(self.get_train_dataloader())
-        self.nb_training_batches = int(self.nb_training_batches * self.train_percent_check)
+        if isinstance(self.get_train_dataloader(), IterableDataset):
```

Comment:
Looks like this needs to be `isinstance(self.get_train_dataloader().dataset, IterableDataset)`